arch/src/lib.rs

```diff
@@ -12,41 +15,38 @@ extern crate memory_model;
 use std::fmt;
 use std::result;
 
-#[derive(Debug)]
-pub enum Error {
-    #[cfg(target_arch = "aarch64")]
-    /// aarch64 specific error triggered during system configuration.
-    Aarch64Setup(aarch64::Error),
-    #[cfg(target_arch = "x86_64")]
-    /// X86_64 specific error triggered during system configuration.
-    X86_64Setup(x86_64::Error),
-}
-pub type Result<T> = result::Result<T, Error>;
-
+/// Module for aarch64 related functionality.
 #[cfg(target_arch = "aarch64")]
 pub mod aarch64;
 
 #[cfg(target_arch = "aarch64")]
 pub use aarch64::{
     arch_memory_regions, configure_system, get_kernel_start, get_reserved_mem_addr,
-    layout::CMDLINE_MAX_SIZE, layout::IRQ_BASE, layout::IRQ_MAX,
+    layout::CMDLINE_MAX_SIZE, layout::IRQ_BASE, layout::IRQ_MAX, Error,
 };
 
+/// Module for x86_64 related functionality.
 #[cfg(target_arch = "x86_64")]
 pub mod x86_64;
 
 #[cfg(target_arch = "x86_64")]
 pub use x86_64::{
     arch_memory_regions, configure_system, get_32bit_gap_start as get_reserved_mem_addr,
-    get_kernel_start, layout::CMDLINE_MAX_SIZE, layout::IRQ_BASE, layout::IRQ_MAX,
+    get_kernel_start, layout::CMDLINE_MAX_SIZE, layout::IRQ_BASE, layout::IRQ_MAX, Error,
 };
 
+/// Type for returning public functions outcome.
+pub type Result<T> = result::Result<T, Error>;
+
 /// Types of devices that can get attached to this platform.
 #[derive(Clone, Debug, PartialEq, Eq, Hash, Copy)]
 pub enum DeviceType {
+    /// Virtio Device Type.
```

Comment:
nit: rephrase for clarity: this is type `virtio` and not the `type of virtio`:
```suggestion
    /// Device Type: Virtio.
```